app/redux/ducks/classify.js

```diff
@@ -141,9 +141,18 @@ const TOGGLE_GOLD_STANDARD = 'pfe/classify/TOGGLE_GOLD_STANDARD';
 export default function reducer(state = initialState, action = {}) {
   switch (action.type) {
     case ADD_INTERVENTION: {
+      // this action payload is the json intervention object from sugar
       const intervention = action.payload;
       const { classification } = state;
-      if (classification && classification.links.project === intervention.project_id) {
+      const { project, workflow } = classification.links;
+      const matchesProject = project === String(intervention.project_id);
+      // only test workflow matches known state if the incoming payload has this property
+      // to allow project level as well as targetted workflow messages
+      let matchesWorkflow = true;
+      if (intervention.hasOwnProperty('workflow_id')) {
```

Comment:
Is there any chance `intervention` will be undefined here?

Reply:
`hasOwnProperty` will evaluate whether or not that object has that property, but not if the property has a value, so: 
```
const intervention = { workflow_id: undefined }
intervention.hasOwnProperty('workflow_id') // true
```

If you mean to check that the property exists and has a value, I think you'll have to use dot notation:
```
if (intervention && intervention.workflow_id) {
 // do something
}
```